src/torcontrol.h

```diff
@@ -11,18 +11,14 @@
 #include <netaddress.h>
 #include <util/fs.h>
 
+#include <event2/util.h>
 
-#include <event2/bufferevent.h>
-#include <event2/event.h>
```

Comment:
Is this (another) a bug in IWYU? Seems a bit weird to remove the `<event2/event.h>` include, which, as far as I can tell, is correct. It's where `event_base` is defined, and what is [expected when using libevent](https://github.com/libevent/libevent/blob/c9ec6aafb6a025f03636ae2ae7c18a2d4b8a7ed8/include/event2/event.h#L55):
>   section usage Standard usage
>  Every program that uses Libevent must include the <event2/event.h>
  header, and pass the -levent flag to the linker.

Reply:
I think the since the libevent structs in this file are only pointers, the actual definitions aren't needed hence the include is not strictly necessary?

Reply:
Yeah, iirc everything in libevent can be forward-declared except for `evutil_socket_t`. It's very annoying.

Because everything else is already forward-declared as @achow101 mentioned, only the definition of `evutil_socket_t` is needed. `evutil_socket_t` is actually defined in `event2/util.h` (note that `event.h` includes `util.h`.

So I agree with this change. It's a small reduction in needed includes.